core/src/main/kotlin/com/github/inflab/spring/data/mongodb/core/aggregation/ReplaceRootDsl.kt

```diff
@@ -0,0 +1,173 @@
+package com.github.inflab.spring.data.mongodb.core.aggregation
+
+import com.github.inflab.spring.data.mongodb.core.aggregation.expression.AggregationExpressionDsl
+import com.github.inflab.spring.data.mongodb.core.annotation.AggregationMarker
+import com.github.inflab.spring.data.mongodb.core.extension.toDotPath
+import org.springframework.data.mongodb.core.aggregation.AggregationExpression
+import org.springframework.data.mongodb.core.aggregation.ReplaceRootOperation
+import org.springframework.data.mongodb.core.aggregation.ReplaceRootOperation.ReplaceRootDocumentOperation
+import kotlin.reflect.KProperty
+
+/**
+ * A Kotlin DSL to configure $replaceRoot stage using idiomatic Kotlin code.
+ *
+ * @author username1103
+ * @since 1.0
+ * @see <a href="https://www.mongodb.com/docs/manual/reference/operator/aggregation/replaceRoot">$replaceRoot (aggregation)</a>
+ */
+@AggregationMarker
+class ReplaceRootDsl {
+    private var operation: ReplaceRootOperation? = null
+
+    /**
+     * Specifies the new root document with document operation.
+     *
+     * @param configuration The configuration for the [ReplacementDocumentOperationDsl]
+     */
+    fun newRoot(configuration: ReplacementDocumentOperationDsl.() -> Unit) {
+        operation = ReplacementDocumentOperationDsl().apply(configuration).get()
+    }
+
+    /**
+     * Specifies field name for the new root document.
+     *
+     * @param path The path of the field to replace new root.
+     */
+    fun newRoot(path: String) {
+        operation = ReplaceRootOperation.builder().withValueOf(path)
+    }
+
+    /**
+     * Specifies field name for the new root document.
+     *
+     * @param path The path of the field to replace new root.
+     */
+    fun newRoot(path: KProperty<*>) {
+        operation = ReplaceRootOperation.builder().withValueOf(path.toDotPath())
+    }
+
+    /**
+     * Specifies the new root document with aggregation expression.
+     *
+     * @param configuration The configuration block where you can use DSL to define aggregation expression.
+     */
+    fun expressions(configuration: AggregationExpressionDsl.() -> AggregationExpression) {
+        operation = ReplaceRootOperation.builder().withValueOf(AggregationExpressionDsl().configuration())
+    }
+
```

Comment:
I want to overloading for function`newRoot`. Like that
```kt
 fun newRoot(configuration: AggregationExpressionDsl.() -> AggregationExpression) {
    operation = ReplaceRootOperation.builder().withValueOf(AggregationExpressionDsl().configuration())
}
```

But raise `Overload resolution ambiguity` from `ReplacementDocumentOperationDsl` and this. 

So How about that function name use `document`, `field`, and `expressions`?
Or is there a good way?

Reply:
How about adding `prefix: Stirng` argument for newRoot for `ReplacementDocumentOperationDsl` ?